nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/RouteText.java

```diff
@@ -655,14 +653,9 @@ public boolean equals(Object obj) {
 
             Group other = (Group) obj;
             if (capturedValues == null) {
-                if (other.capturedValues != null) {
-                    return false;
-                }
-            } else if (!capturedValues.equals(other.capturedValues)) {
-                return false;
-            }
+                return other.capturedValues == null;
+            } else return capturedValues.equals(other.capturedValues);
```

Comment:
This was an automatic IDE optimization I wasn't aware of, but the logic checks out.